WorkloadManagementSystem/Executor/JobScheduling.py

```diff
@@ -222,16 +222,12 @@ def optimizeJob( self, jid, jobState ):
     return self.__setJobSite( jobState, stageSites )
 
   def __applySiteFilter( self, sites, active = False, banned = False ):
-    filtered = list( sites )
-    if active:
-      for site in sites:
-        if site not in active:
-          filtered.remove( site )
-    if banned:
-      for site in banned:
-        if site in filtered:
-          filtered.remove( site )
-    return filtered
+    filtered = set( sites )
+    if active and isinstance( active, ( list, set, dict ) ):
+      filtered &= set( active )
+    if banned and isinstance( banned, ( list, set, dict ) ):
+      filtered -= set( banned )
+    return list( filtered )
 
```

Comment:
... I didn't re-write it recently, it is as was originally in the PR...
Still Federico sorry but I don't understand your line of code... For me this is not doing anything, as filtered is unaffected??? You subtract the sites requested by the user (why?) and then you OR it with filtered which results in filtered = filtered, right?